lib/console/console.c

```diff
@@ -637,7 +654,23 @@ static status_t command_loop(console_t *con, int (*get_line)(const char **, void
             mutex_acquire(&con->lock);
 
         con->abort_script = false;
-        con->lastresult = command->cmd_callback(argc, args);
+        if (strcmp(args[argc - 1].str, "&") == 0) { // background execution
+            background_cmd_args *bg_args =
+                (background_cmd_args *)malloc(sizeof(background_cmd_args));
+            bg_args->command = command;
+            bg_args->argc = argc - 1;
+            bg_args->args = (console_cmd_args *)malloc(argc * sizeof(console_cmd_args));
+            for (int i = 0; i < argc; i++) {
+                memcpy(&bg_args->args[i], &args[i], sizeof(console_cmd_args));
+                bg_args->args[i].str = strdup(args[i].str);
+            }
+            thread_t *thr = thread_create("background_command_exec", background_command_exec,
+                        bg_args, DEFAULT_PRIORITY,
+                        DEFAULT_STACK_SIZE);
+            con->lastresult = thread_resume(thr);
```

Comment:
suggestion: make this thread_detach_and_resume, instead of thread_resume. This allows it to exit on its own, without requiring the original thread needing to thread_join() it.